crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -761,6 +769,18 @@ impl<'src> Parser<'src> {
                 return lower.expr;
             }
 
+            if !lower.is_parenthesized {
+                match lower.expr {
```

Comment:
Longterm, I think we want to move this checks into `checker`. Once we have something like that ;)